File: 1381. Design a Stack With Increment Operation/Solution.cpp
class CustomStack {
public:
    int max;
    std::vector<int> stack;
        
    CustomStack(int maxSize) {
        this->max = maxSize;
        this->stack.reserve(maxSize);
    }
    
    void push(int x) {
        if (this->stack.size() < this->max)
        {
            this->stack.push_back(x);
        }        
    }
    
    int pop() {
        if (this->stack.size() > 0)
        {
            int ret = this->stack[this->stack.size()-1];
            stack.pop_back();
            return ret;
        }
        else
        {
            return -1;
        }
    }
    
    void increment(int k, int val) {
        for(int i=0;i<k;i++)
        {
            if (i < this->stack.size())
            {
                this->stack[i] += val;
            }
            else
            {
                break;
            }
        }
    }
};

/**
 * Your CustomStack object will be instantiated and called as such:
 * CustomStack* obj = new CustomStack(maxSize);
 * obj->push(x);
 * int param_2 = obj->pop();
 * obj->increment(k,val);
 */

File: 1381. Design a Stack With Increment Operation/Solution.py
class CustomStack:

    def __init__(self, maxSize: int):
        self.maxSize = maxSize
        self.stack = []
        

    def push(self, x: int) -> None:
        if len(self.stack) < self.maxSize:
            self.stack.append(x)

    def pop(self) -> int:
        try:
            return self.stack.pop()
        except:
            return -1
        

    def increment(self, k: int, val: int) -> None:
        for i in range(k):
            try:
                self.stack[i] += val
            except:
                break
        


# Your CustomStack object will be instantiated and called as such:
# obj = CustomStack(maxSize)
# obj.push(x)
# param_2 = obj.pop()
# obj.increment(k,val)
